eng/tools/compare-packages.ps1

```diff
@@ -0,0 +1,47 @@
+param (
+  $pathToMasterPkg,
+  $pathToCurrentPkg,
+  $dailyDevBuildNo,
+  $pathForDiffFile
+)
+
+function ExtractTGZPackages($pathToPkg){
+  $regExp = "-\d+\.\d+\.\d+(-\w*\.\d*)?(-dev-$dailyDevBuildNo)?"
+  foreach ($p in $(dir $pathToPkg -r -i *.tgz)){
+    $extractDir = (get-item $p.FullName).Directoryname + "/" + $p.BaseName
+    if($p.BaseName -match $regExp){
+      $newDirName = ($p.BaseName -split $Matches.0)[0]
+      $extractDir = (get-item $p.FullName).Directoryname + "/" + $newDirName
+    }
+    mkdir "$extractDir"
+    pushd "$extractDir"
+    tar -xzf $p.FullName
+    popd
+  }
+}
+try{
+  ExtractTGZPackages($pathToMasterPkg)
+  if($LastExitCode -ne 0){
+    Write-Host "error >> ExtractTGZPackages($pathToMasterPkg) failed with exit code $LastExitCode"
+  }
+
+  ExtractTGZPackages($pathToCurrentPkg)
+  if($LastExitCode -ne 0){
+    Write-Host "error >> ExtractTGZPackages($pathToCurrentPkg) failed with exit code $LastExitCode"
+  }
+  echo "Finished unzipping of all .tgz packages"
+
+  $diffFile = $pathForDiffFile+"/Change_"+$dailyDevBuildNo + ".diff"
+  echo "created filename variable $diffFile"
+
+  git diff $pathToMasterPkg $pathToCurrentPkg | tee $diffFile
```

Comment:
I modified the script to use Join-Path and now I am creating a subfolder inside $pathtoPkg named "all-contents" where the packages are extracted and that is the subfolder I will use to diff so that it doesn't include binary ".tgz" or ".zip" files in the report